### PR TITLE
Web apps: hide pagination controls when unnecessary

### PR DESCRIPTION
**Pull request by orangejenny**
## Product Description
https://dimagi-dev.atlassian.net/browse/USH-3936

### Safety story
Impact should be limited to pagination, though web apps js changes can always break web apps.

### Automated test coverage

in PR

### QA Plan

https://dimagi-dev.atlassian.net/browse/QA-5976

### Rollback instructions

- [x] This PR can be reverted after deploy with no further considerations

### Labels & Review
- [x] Risk label is set correctly
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change

### CODE REVIEW

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js [hunk 1]
```diff
@@ -160,12 +161,18 @@ hqDefine("cloudcare/js/formplayer/utils/utils", function () {
                 endPage = currentPage + maxPagesAfterCurrentPage;
             }
         }
+
+        let rowRange = [5, 10, 25, 50, 100],
+            hasMultiplePages = totalPages > 1,
+            hasAtLeastMinimumItemsPerPage = totalItems > rowRange[0],
+            showPagination = hasMultiplePages || hasAtLeastMinimumItemsPerPage;
```
Comment (nospame) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js:168]: I think these can be `const` unless they need to be redefined later.

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/case_list_pagination_spec.js [hunk 2]
```diff
@@ -2,7 +2,7 @@ hqDefine("cloudcare/js/formplayer/spec/case_list_pagination_spec", function () {
     describe('#paginateOptions', function () {
         var paginateItems = hqImport("cloudcare/js/formplayer/utils/utils");
         it('Should return paginateOptions', function () {
-            var case1 = paginateItems.paginateOptions(0, 15);
+            var case1 = paginateItems.paginateOptions(0, 15, 3);
```
Comment (nospame) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/case_list_pagination_spec.js:5]: Should there be `showPagination` assertions for `case1` and `case4` in this block as well?

**Review by nospame — APPROVED** [review]
This looks good to me, couple non-blocking comments. What are the different contexts pagination is removed where unneeded? I think it's case list, users, and one other.

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js (hunk 1 above)
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js]: yes, thanks - 044b0ea89d8

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/case_list_pagination_spec.js (hunk 2 above)
Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/case_list_pagination_spec.js:5]: Meh. It wouldn't test additional behavior.

**Comment by orangejenny** [comment]
> What are the different contexts pagination is removed where unneeded? I think it's case list, users, and one other.

Yep, the two you named and then the incomplete forms screen.

**Review by nospame — APPROVED** [review]

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js [hunk 3]
```diff
@@ -160,10 +161,19 @@ hqDefine("cloudcare/js/formplayer/utils/utils", function () {
                 endPage = currentPage + maxPagesAfterCurrentPage;
             }
         }
+
+        const rowRange = [5, 10, 25, 50, 100],
+            hasMultiplePages = totalPages > 1,
+            hasAtLeastMinimumItemsPerPage = totalItems > rowRange[0],
```
Comment (MartinRiese) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js:167]: when would `hasAtLeastMinimumItemsPerPage` be true but `hasMultiplePages` not?

Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js:167]: Say the user has selected 10 items per page, and there are 8 items: `totalPages` would be 1, but `totalItems` is greater than the bare minimum (5).

Comment (MartinRiese) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js:167]: But why would you have to show pagination in that case? wouldn't everything be on the one page?

Comment (orangejenny) [reply corehq/apps/cloudcare/static/cloudcare/js/formplayer/utils/utils.js:167]: It gives the user the option to decrease the number of items per page - they'll only see that dropdown, there are already other checks that hide the buttons to jump to different pages if there's only one page.

**Pull request merged**